client/main.lua

```diff
@@ -179,68 +221,173 @@ CreateThread(function()
+                            return true
+                        else
+                            return false
+                        end
+                    end,
+                  }
+            }
+            if Config.WeaponRepairPoints[i].target.usePed and not pedsSpawned then
+                local model = Config.WeaponRepairPoints[i].target.pedModel
+                RequestModel(model)
+                while not HasModelLoaded(model) do
+                  Wait(0)
+                end
+                local pos = Config.WeaponRepairPoints[i].coords
+                local entity = CreatePed(0, model, pos.x, pos.y, pos.z-1, pos.w, false, false)
+                SetBlockingOfNonTemporaryEvents(entity, true)
+                FreezeEntityPosition(entity, true)
+                SetEntityInvincible(entity, true)
+                exports['qb-target']:AddTargetEntity(entity, {
+                  options = opts,
+                  distance = 2.5,
+                })
+            else
+                exports['qb-target']:AddBoxZone("weaponrepair"..i, vector3(v.coords.x, v.coords.y, v.coords.z), v.target.width, v.target.depth, {
+                    name = "weaponrepair"..i,
+                    heading = v.coords.w,
+                    debugPoly = v.target.debug,
+                    minZ = v.target.minZ,
+                    maxZ = v.target.maxZ,
+                  },{
+                    options = opts,
+                    distance = 2.5,
+                })
+            end
+        end
+        pedsSpawned = true
+    end)
+else
+    CreateThread(function()
+        while true do
+            if LocalPlayer.state.isLoggedIn then
+                local inRange = false
+                local ped = PlayerPedId()
+                local pos = GetEntityCoords(ped)
+                for k, data in pairs(Config.WeaponRepairPoints) do
+                    local distance = #(pos - vector3(data.coords.x, data.coords.y, data.coords.z))
+                    if distance < 10 then
+                        inRange = true
+                        if distance < 1 then
+                            if data.IsRepairing then
+                                if data.RepairingData.CitizenId ~= PlayerData.citizenid then
+                                    DrawText3Ds(data.coords.x, data.coords.y, data.coords.z, Lang:t('info.repairshop_not_usable'))
                                 else
-                                    DrawText3Ds(data.coords.x, data.coords.y, data.coords.z, Lang:t('info.take_weapon_back'))
+                                    if not data.RepairingData.Ready then
+                                        DrawText3Ds(data.coords.x, data.coords.y, data.coords.z, Lang:t('info.weapon_will_repair'))
+                                        if IsControlJustPressed(0, 38) then
+                                            local data = {id = k}
+                                            TriggerServerEvent('weapons:server:RepairTime', data)
```

Comment:
```suggestion
                                            TriggerServerEvent('weapons:server:RepairTime', {id = k})
```

Resolve linting issues